util/pyclient/bft_client.py

```diff
@@ -163,14 +163,51 @@ async def sendSync(self, msg, read_only, seq_num=None, cid=None, pre_process=Fal
         # Raise a trio.TooSlowError exception if a quorum of replies
         try:
             with trio.fail_after(self.config.req_timeout_milli / 1000):
-                self._reset_on_new_request()
-                return await self._send_receive_loop(data, read_only, m_of_n_quorum)
+                self._reset_on_new_request([seq_num])
+                replies = await self._send_receive_loop(data, read_only, m_of_n_quorum)
+                return next(iter(self.replies.values())).get_common_data()
         except trio.TooSlowError:
             print("TooSlowError thrown from client_id", self.client_id, "for seq_num", seq_num)
             raise trio.TooSlowError
         finally:
             pass
 
+    async def write_batch(self, msg_batch, batch_seq_nums=None, m_of_n_quorum=None):
+        if not self.comm_prepared:
+            await self._comm_prepare()
+
+        cid = str(self.req_seq_num.next())
+
+        if batch_seq_nums is None:
+            batch_seq_nums = []
```

Comment:
Modifying a parameter isn't very clean. Please use a local variable instead.

Reply:
The same approach is used for non-required parameters in bft_client. I think we should keep it like this for consistency.

Reply:
Agreed, makes sense.